packages/lisk-transaction-pool/test/integration/transaction_movement.ts

```diff
@@ -0,0 +1,183 @@
+            transactions.forEach(transaction => {
+                transactionPool.addTransaction(transaction);
+            });
+            transactionsToValidate = transactionPool.queues.received.peekUntil(returnTrueUntilLimit(configuration.receivedTransactionsLimitPerProcessing));
+            const { passedTransactions, failedTransactions } = validateTransactionFunction(transactionsToValidate);
+            
+            validTransactions = passedTransactions;
+            invalidTransactions = failedTransactions;
+            clock.tick(configuration.receivedTransactionsProcessingInterval + 1);
+        });
+
+        it('should remove transactions from the received queue', async () => {
+            return wrapExpectationInNextTick(() => {
+                transactionsToValidate.forEach(transaction => {
+                    expect(transactionPool.queues.received.exists(transaction)).to.equal(false);
+                });
+            });
+        });
+
+        it('should move valid transactions to the validated queue', async () => {
+            return wrapExpectationInNextTick(() => {
+                validTransactions.forEach(transaction => {
+                    expect(transactionPool.queues.validated.exists(transaction)).to.equal(true);
+                });
+                expect(transactionPool.queues.validated.size()).to.equal(validTransactions.length);
+            });
+        });
+
+        it('should remove invalid transactions from the transaction pool', async () => {
+            return wrapExpectationInNextTick(() => {
+                invalidTransactions.forEach(transaction => {
+                    expect(transactionPool.existsInTransactionPool(transaction)).to.equal(false);
+                });
+            });
+        });
+
+        describe('from validated to the verified and pending queues', () => {
+            let transactionsToVerify: ReadonlyArray<Transaction>;
+            let verifiableTransactions: ReadonlyArray<Transaction>;
+            let unverifiableTransactions: ReadonlyArray<Transaction>;
+
+            beforeEach(async () => {
+                transactionsToVerify = validTransactions;
+                const { passedTransactions, failedTransactions } = verifyTransactionFunction(transactionsToVerify);
+
+                verifiableTransactions = passedTransactions;
+                unverifiableTransactions = failedTransactions;
+                clock.tick(configuration.validatedTransactionsProcessingInterval + 1);
+            });
+
+            it('should remove transactions from the validated queue', async () => {
+                return wrapExpectationInNextTick(() => {
+                    transactionsToVerify.forEach(transaction => {
+                        expect(transactionPool.queues.validated.exists(transaction)).to.equal(false);
+                    });
+                });
+            });
+
+            it('should move verifiable transactions to the verfied queue', async () => {
```

Comment:
im not sure `verifiable` is the right word here @mitsujutsu ?

Reply:
just `verified` should be ok?